doc/guides/modules-in-core.md

```diff
@@ -0,0 +1,44 @@
+# To be or not to be in core
+
+Should a module be in core? This question arises every so often. This document
+explains things to consider when deciding whether a module should be in core or
+not.
+
+## Strong arguments for including a module in core
+
+1. The module provides functionality that is standardized (such as a
+   [Web API][]) and overlaps with existing functionality.
+2. The module can only be implemented in core.
+3. The module can only be implemented in a performant way in core.
+4. Developer experience is significantly improved if the module is in core.
+5. The module will be widely used, have cross platform native code, and being in
+   core will avoid the need for an end-user compilation chain.
+6. Part or all of the module will also be re-used or duplicated in core.
+
+## Strong arguments against including a module in core
```

Comment:
Other strong arguments against including a module in core:

 - The module has a license that prohibits Node.js from including it in core without also changing its own license.
 - There is already similar functionality in core and adding the module will provide a second API to do the same thing.
 - A module (or/and the standard it is based on) is deprecated and there is a non-deprecated alternative.
 - The module is evolving quickly and inclusion in core will require frequent API changes.

?